MLSDK-348: Surface status endpoint

DanielTamkin opened the pull request:
## Summary

Introduces the command `hyper remoteStatus`.

### `hyper remoteStatus` : start a status endpoint for polling

Summons the status endpoint, if the specified port is unavailable the command will fail. Ping `localhost:3001/status` to receive the status in JSON, example:

```json
{
  "message": "Pulling docker image"
}
```

If the `message` is an empty string, or the statusFile does not exist, The response will be a [204](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204).



**Optional flags:**

- `--port`: _(Default: `3001`)_ Specify the custom port to launch the endpoint with
- `--statusFile`: _(Default: `./statusfile.json`)_ Specify the location of the status file.

### `hyper remoteStatus update "<message>"` : Update the status file

Updates the status file with the supplied `message` in JSON format.

**Optional flags:**

- `--statusFile`: _(Default: `./statusfile.json`)_ Specify the location of the status file.



## Ticket(s)

- [MLSDK-348](https://gohypergiant.atlassian.net/browse/MLSDK-348)
- [MLSDK-347](https://gohypergiant.atlassian.net/browse/MLSDK-347)
- [MLSDK-346](https://gohypergiant.atlassian.net/browse/MLSDK-346)

DanielTamkin committed:
feat(cmd): add `remoteStatus` command

With flags `--port` & `--statusFile`, defaulting to 3001 & ./statusFile.json - respectively.

Enables:
* `hyper remoteStatus` to start the status endpoint
* `hyper remoteStatus update "<message>"` to update the status file

DanielTamkin committed:
chore: ignore `statusfile.json`

DanielTamkin committed:
docs(README): include `hyper remoteStatus` documentation

DanielTamkin committed:
feat(client): include `hyper remoteStatus` in `EC2StartScript`s

DanielTamkin committed:
chore: cleanup

DanielTamkin committed:
Merge branch 'stable' into feature/MLSDK-348-surface-status-endpoint

DanielTamkin committed:
Merge branch 'stable' into feature/MLSDK-348-surface-status-endpoint

craigjmidwinter commented on hyper/client/aws/ec2.go:
```diff
@@ -658,11 +658,13 @@ mkdir -p /tmp/hyperdrive/project
 curl -fsSL https://github.com/gohypergiant/hyperdrive/releases/download/%s/hyperdrive_%s_Linux_x86_64.tar.gz -o /tmp/hyperdrive/hyper.tar
 tar -xvf /tmp/hyperdrive/hyper.tar -C /tmp/hyperdrive
 mv /tmp/hyperdrive/hyper /usr/bin/hyper
+hyper remoteStatus &
```
The one other change that we'll want to make to this file before we merge is to bump the version string to the current version+1 so that when the startup script pulls the binary it pulls one that has the `remoteStatus` subcommand

craigjmidwinter commented:
Looks good, just one small addition

DanielTamkin committed:
Merge branch 'stable' into feature/MLSDK-348-surface-status-endpoint

DanielTamkin committed:
chore: bump `version` string to `0.0.58`

Allows the startup script to pull the latest binary with this `remoteStatus` command.

DanielTamkin committed:
Merge branch 'stable' into feature/MLSDK-348-surface-status-endpoint

craigjmidwinter reviewed (approved):
LGTM!

DanielTamkin committed:
chore: bump `version` to `0.0.59`
